go/test/endtoend/reparent/main_test.go

```diff
@@ -0,0 +1,175 @@
+/*
+Copyright 2019 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package reparent
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"os"
+	"os/exec"
+	"path"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"vitess.io/vitess/go/mysql"
+	"vitess.io/vitess/go/sqltypes"
+	"vitess.io/vitess/go/test/endtoend/cluster"
+	tmc "vitess.io/vitess/go/vt/vttablet/grpctmclient"
+)
+
+var (
+	// ClusterInstance instance to be used for test with different params
+	clusterInstance *cluster.LocalProcessCluster
+	tmClient        *tmc.Client
+	keyspaceName    = "ks"
+	shardName       = "0"
+	shard1Name      = "0000000000000000-ffffffffffffffff"
```

Comment:
No, copied from Python ones. Will test with `0-f`

Reply:
Updated it to `-80`. Let me know if that works for the test

Reply:
I looked at the original python test case. I think this does need to be the same as it was in python.
Also, in the python test, we basically run the same scenario for shard0 and shard1, the only difference is how the shard is represented: `0` vs `0000...-ffff...`. We should preserve that while porting rather than changing the test to only have 1 master + 1 replica when the shard is represented as `0000...-ffff...`

Reply:
I have reverted the range to `000...-ffff...`.
As wrt to replica counts I have only used master-replica as we just need to verify replication in a replica in range-based test. 
As opposed to other Tests like `TestChangeTypeSemiSync` and  `TestReparentFromOutside` need 4 replicas setup which is already present in Shard0.